server/app/models/distributed_lock.rb

```diff
@@ -13,17 +15,16 @@ def self.with_lock(name, timeout = 60)
     lock_id = nil
     begin
       if timeout.to_f > 0.0
-        Timeout.timeout(timeout) do
-          sleep 0.01 until lock_id = self.obtain_lock(name)
-        end
+        lock_id = WaitHelper.wait(timeout: timeout) { self.obtain_lock(name) }
```

Comment:
I just commented on this in the main PR: https://github.com/kontena/kontena/pull/1987/files#r106355611